packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -250,7 +250,13 @@ export const createTransactionsTracker = (
   );
 
   const txOnChain$ = (evt: OutgoingTx): Observable<OutgoingOnChainTx> =>
-    onChainNewTxSuccess$.pipe(
+    merge(
+      historicalTransactions$.pipe(
+        take(1),
+        mergeMap((txs) => from(txs))
+      ),
+      onChainNewTxSuccess$
```

Comment:
I think it's safe to remove onChainNewTxSuccess$. It is included in historicalTransactions$

Reply:
It has a `take(1)` from `historicalTransactions$`, so that we don't have to check it against every historical transaction twice.

Reply:
Missed that. good to go 😄